### ISSUE TITLE
Why does the loss/alpha of my SAC algorithm not converge after changing the graphics card?

### ISSUE THREAD

**Issue by jaried** [aai-institute/tianshou#590]
-  I have marked all applicable categories:

-  I have visited the [source website](https://github.com/thu-ml/tianshou/)
-  I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
-I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
0.4.7 0.21.0 1.11.0+cu113 1.20.3 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)] win32
  ```
My graphics card changed from 970 to 3090 TI, the code has not changed, but, from the tensorboard, the loss/alpha has not converged. The red line is the result of training with 970 before, and the sky blue line is the result of training with 3090 Ti. , what could be the cause?

I have tried modifying the seed, retraining from 0, and so on, and the results are the same.
![1](https://user-images.githubusercontent.com/26672425/162619179-3bb49281-5c1f-45e1-89dd-1d560b85bd3c.png)

**Comment by Trinkle23897** [comment]
I don't think the blue curve is "wrong". It's quite normal for SAC algorithm.

**Comment by jaried** [comment]
But from the perspective of reward, the corresponding reward in red is normal.

**Comment by Trinkle23897** [comment]
Could you please also put the plot of reward here?

**Comment by jaried** [comment]
Color changed.

![1](https://user-images.githubusercontent.com/26672425/162620939-56af8fbf-438e-4dac-a24f-87d5217f12b0.png)

**Comment by Trinkle23897** [comment]
Hmm... not sure what happens but my guess is 970's driver is not compatible with the recent torch version, thus causing such a difference.

BTW, is 970's result still producing all 0? How about 3090ti?

**Comment by jaried** [comment]
I have reinstalled the 3090ti driver, cuda, cudnn, just reinstalled pytorch1.11 and try again.

The orange line is 970 , the blue is 3090 ti, I used logarithmic coordinates for positive coordinates.
![1](https://user-images.githubusercontent.com/26672425/162621223-07f700ac-fa64-434a-9778-3f4cc0adfbbb.png)


The following figure is a common coordinate system:
Losses are first divergent and then convergent.
![2](https://user-images.githubusercontent.com/26672425/162622242-1d35e66e-07f6-4316-b85e-18ed98877bce.png)

**Comment by jaried** [comment]
Orange is the ideal state of the previous 970, red is the result of the 3090ti training last night, and it still hasn't converged, so try switching back to the 970 another day.
![1](https://user-images.githubusercontent.com/26672425/162646698-6b4b5ab1-a976-4818-abc9-74d115c6f18d.png)

Below is the normal coordinate system.
![2](https://user-images.githubusercontent.com/26672425/162646766-9b70805d-9373-4e37-8021-3be35c7d000b.png)

**Comment by Trinkle23897** [comment]
Could you please check out the most recent commit and rerun it?

**Comment by jaried** [comment]
I used the history function of pycharm to restore the code to before changing the graphics card. Now I just changed the graphics card back to 970. To see the effect of training tonight, it usually takes 8 hours of training. Under the 3090ti, I have tried all the methods I know, and the reward has never been able to reach the level of the previous 970.

**Comment by jaried** [comment]
After checking it, I accidentally found that the environment has fixed the seed to a certain value, and did not change the graphics card. It's Weng Da, you asked me to write all the environments to be consistent, so I also fixed the seed to a certain value. But then it didn't change back.

I didn't get great training results with the 3090Ti last night, so I switched back to the 970 to continue training...
